compiler/rustc_mir_build/src/build/expr/as_rvalue.rs

```diff
@@ -436,9 +436,9 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             Rvalue::Ref(this.hir.tcx().lifetimes.re_erased, borrow_kind, arg_place),
         );
 
-        // In constants, temp_lifetime is None. We should not need to drop
-        // anything because no values with a destructor can be created in
-        // a constant at this time, even if the type may need dropping.
+        // Certain dead code around "values" of `!` type will not actually have a lifetime
+        // associated with them.
+        // FIXME: can we generate one?
         if let Some(temp_lifetime) = temp_lifetime {
             this.schedule_drop_storage_and_value(upvar_span, temp_lifetime, temp);
         }
```

Comment:
This comment was copied from here (as part of #52405's c3dbdfee06c733f36420a651cd9682e0ee4f95c3): https://github.com/rust-lang/rust/blob/c3dbdfee06c733f36420a651cd9682e0ee4f95c3/src/librustc_mir/build/expr/as_temp.rs#L61-L68

This is the current version, updated in (your) #56127: https://github.com/rust-lang/rust/blob/e0ef0fc392963438af5f0343bf7caa46fb9c3ec3/compiler/rustc_mir_build/src/build/expr/as_temp.rs#L98-L113

If we don't want to copy the comment (which is pretty long), we could at least point to the other one?
Or if that's not applicable, it could still mention `as_temp` but say that the same situation does not arise for what's handled here.